src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt

```diff
@@ -594,9 +595,9 @@ class SnykToolWindowPanel(val project: Project) : JPanel(), Disposable {
             settings.pluginFirstRun -> {
                 pluginSettings().pluginFirstRun = false
                 enableProductsAccordingToServerSetting()
-                getSyncPublisher(project, SnykSettingsListener.SNYK_SETTINGS_TOPIC)?.settingsChanged()
                 displayTreeAndDescriptionPanels()
-                triggerScan()
+                // don't trigger scan for Default project i.e. no project opened state
+                if (project.basePath != null) triggerScan()
```

Comment:
It would be great to have this tested, if it's possible.

Reply:
I'm afraid writing tests for that will take 10 times longer then actual fix :(
That might be luck of my testing experience... but here particularly I would either drop the PR then wasting time for writing tests...

Reply:
Agree, a test would be nice. Would probably involve a few mocks, so yeah, I understand the reluctance. Still, would be great :)

Reply:
@ArtsiomCh I feel you here. 

@bastiandoetsch I think we should invest more time into our architecture to enable unit testing of such small bits.

Reply:
For sure. SnykToolWindowPanel needs a thorough decomposition, it is way too complex and needs to be split-up.